manifestival.go

```diff
@@ -103,20 +104,26 @@ func (m Manifest) apply(spec *unstructured.Unstructured, opts ...ApplyOption) er
 		annotate(current, "manifestival", resourceCreated)
 		return m.Client.Create(current, opts...)
 	} else {
-		if ApplyWith(opts).Overwrite {
-			return m.update(spec.DeepCopy(), lastApplied(spec), opts...)
-		}
-		diff, err := patch.New(current, spec)
-		if err != nil {
-			return err
-		}
-		if diff != nil {
+		if !ApplyWith(opts).Overwrite {
+			diff, err := patch.New(current, spec)
+			if err != nil {
+				return err
+			}
+			if diff == nil {
+				return nil
+			}
 			m.log.Info("Merging", "diff", diff)
 			if err := diff.Merge(current); err != nil {
 				return err
 			}
-			return m.update(current, lastApplied(spec), opts...)
+			err = m.update(current, lastApplied(spec), opts...)
+			if err == nil {
+				return nil
+			}
```

Comment:
I wonder that it is better to filter the error type (like duplicated value error only), but it is not possible?

Reply:
It is possible, sure. I can recognize any field error by its type, e.g. https://github.com/kubernetes/apimachinery/blob/kubernetes-1.16.0/pkg/util/validation/field/errors.go#L103

I figured any error is worth a retry, but I could see how retrying a network error would be useless. Do you think maybe only retry field/Errors? https://github.com/kubernetes/apimachinery/blob/kubernetes-1.16.0/pkg/util/validation/field/errors.go#L29-L36

Reply:
Great. Yes, I thought that network error is useless. So only retry field/Errors would be better, I think.

Reply:
Cool. Done. We only retry for validation errors now. This cleaned up the logic quite a bit, and the `Overwrite` option now works like the `kubectl apply --overwrite` option, which also defaults to true.

Thanks for the feedback!